### ISSUE TITLE
Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.

### ISSUE THREAD

**Issue by liudonghua123** [JulianG/react-list-drag-and-drop#4]
My partial code was something like the belows, it works for display and addition, deletion but not works for dragging and dropping.
```

  handleRLDDChange = (reorderedItems) => {
    this.setState({ items: reorderedItems });
  };

  itemRenderer = (item, index) => {
    return (
      <Card className={this.props.classes.card} key={`${item.search}-${item.replace}`}>
        <CardContent className={this.props.classes.cardContent}>
          <div className={this.props.classes.cardLeftContent}>
            <div className={this.props.classes.cardSearch}>{item.search}</div>
            <div className={this.props.classes.cardArrow}>{' -> '}</div>
            <div className={this.props.classes.cardReplace}>{item.replace}</div>
          </div>
          <div className={this.props.classes.cardRightAction}>
            <Button variant="contained" color="primary" onClick={this.handleDelete(item)}>删除</Button>
          </div>
        </CardContent>
      </Card>
    );
  };

  render() {
    const { classes } = this.props;
    return (
        <div>
          <RLDD
            cssClasses="list-container"
            items={this.state.searchMode ? this.state.searchItems : this.state.items}
            itemRenderer={this.itemRenderer}
            onChange={this.handleRLDDChange}
          />
        </div>
    );
  }
```

I got the following error.
```
The above error occurred in the <RLDDFloatingItemComponent> component:
    in RLDDFloatingItemComponent (created by RLDD)
    in div (created by RLDD)
    in RLDD (at Configuration.js:205)
    in div (at Configuration.js:204)
    in div (at Configuration.js:171)
    in Configuration (created by WithStyles(Configuration))
    in WithStyles(Configuration) (created by Route)
    in Route (at App.js:39)
    in div (at App.js:37)
    in div (at App.js:35)
    in Router (created by HashRouter)
    in HashRouter (at App.js:34)
    in App (created by WithStyles(App))
    in WithStyles(App) (at withRoot.js:35)
    in MuiThemeProvider (at withRoot.js:32)
    in WithRoot (at src/index.js:12)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
function.console.(anonymous function) @ index.js:1452
logCapturedError @ react-dom.development.js:15845
logError @ react-dom.development.js:15880
update.callback @ react-dom.development.js:16716
callCallback @ react-dom.development.js:11636
commitUpdateEffects @ react-dom.development.js:11676
commitUpdateQueue @ react-dom.development.js:11666
commitLifeCycles @ react-dom.development.js:16029
commitAllLifeCycles @ react-dom.development.js:17388
callCallback @ react-dom.development.js:147
invokeGuardedCallbackDev @ react-dom.development.js:196
invokeGuardedCallback @ react-dom.development.js:250
commitRoot @ react-dom.development.js:17549
completeRoot @ react-dom.development.js:19002
performWorkOnRoot @ react-dom.development.js:18924
performWork @ react-dom.development.js:18826
performSyncWork @ react-dom.development.js:18799
requestWork @ react-dom.development.js:18676
scheduleWork @ react-dom.development.js:18480
enqueueSetState @ react-dom.development.js:12143
push../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:386
RLDD._this.handleDragBegin @ RLDD.js:74
(anonymous) @ Signal.js:32
Signal.dispatch @ Signal.js:31
RLDDLogic.handleDragBegin @ RLDDLogic.js:43
RLDDItemComponent.handleMouseMove @ RLDDItemComponent.js:106
react-dom.development.js:57 Uncaught Error: RLDDFloatingItemComponent(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.
    at invariant (react-dom.development.js:57)
    at reconcileChildFibers (react-dom.development.js:13675)
    at reconcileChildren (react-dom.development.js:14081)
    at finishClassComponent (react-dom.development.js:14326)
    at updateClassComponent (react-dom.development.js:14270)
    at beginWork (react-dom.development.js:15106)
    at performUnitOfWork (react-dom.development.js:17903)
    at workLoop (react-dom.development.js:17944)
    at renderRoot (react-dom.development.js:18022)
    at performWorkOnRoot (react-dom.development.js:18919)
    at performWork (react-dom.development.js:18826)
    at performSyncWork (react-dom.development.js:18799)
    at requestWork (react-dom.development.js:18676)
    at scheduleWork (react-dom.development.js:18480)
    at Object.enqueueSetState (react-dom.development.js:12143)
    at RLDD.push../node_modules/react/cjs/react.development.js.Component.setState (react.development.js:386)
    at RLDD._this.handleDragBegin (RLDD.js:74)
    at Signal.js:32
    at Array.forEach (<anonymous>)
    at Signal.dispatch (Signal.js:31)
    at RLDDLogic.handleDragBegin (RLDDLogic.js:43)
    at RLDDItemComponent.handleMouseMove (RLDDItemComponent.js:106)
invariant @ react-dom.development.js:57
reconcileChildFibers @ react-dom.development.js:13675
reconcileChildren @ react-dom.development.js:14081
finishClassComponent @ react-dom.development.js:14326
updateClassComponent @ react-dom.development.js:14270
beginWork @ react-dom.development.js:15106
performUnitOfWork @ react-dom.development.js:17903
workLoop @ react-dom.development.js:17944
renderRoot @ react-dom.development.js:18022
performWorkOnRoot @ react-dom.development.js:18919
performWork @ react-dom.development.js:18826
performSyncWork @ react-dom.development.js:18799
requestWork @ react-dom.development.js:18676
scheduleWork @ react-dom.development.js:18480
enqueueSetState @ react-dom.development.js:12143
push../node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:386
RLDD._this.handleDragBegin @ RLDD.js:74
(anonymous) @ Signal.js:32
Signal.dispatch @ Signal.js:31
RLDDLogic.handleDragBegin @ RLDDLogic.js:43
RLDDItemComponent.handleMouseMove @ RLDDItemComponent.js:106
[Violation] 'mousemove' handler took 83965ms
```

**Comment by liudonghua123** [comment]
@eddiemarovich, I almost gave up about this problem, and I will try another component https://github.com/react-dnd/react-dnd.

**Comment by JulianG** [comment]
Hi @liudonghua123! I'm so sorry, I never got a notification from github about your issue.
I've only seen it today.
I'm trying to understand where your error comes from, but I don't have access to see what your `Card` and `CardContent` components do.

If you haven't given up on this component yet, here's a few things you can try:

- Try rendering something simpler in the itemRenderer function instead of your `<Card>` and `<CardContent>` components. You could start with a `<div>`, or with `<Card>` without `<CardComponent>` and start adding back things until it breaks.
- Alternatively, try to create a demo in CodeSandbox, so I can see the error live.
This is the JS example I made, perhaps you can modify it and add your components there: https://codesandbox.io/s/18050jnp27

**Comment by eddiemarovich** [comment]
Hi Julian, I am also getting the same error and was able to re-create it in the codesandbox example by changing the data that is being rendered to an array of strings, rather than the `bananas.bananas`. I'm not sure why that would be an issue, do you have any insight?

**Comment by eddiemarovich** [comment]
@liudonghua123 @JulianG  I was able to get this working, and I think it had something to do with the the RLDD expecting an object with an ID. @liudonghua123 I'm not sure if you are still trying to mess around with this, but what I did was just created a new array in my constructor and assigned each item I wanted to render an id, and that seems to be working.

**Comment by JulianG** [comment]
@eddiemarovich yes. that is correct.
if you look at the `bananas.json` file in [my example](https://codesandbox.io/s/18050jnp27), every item has an `id` which is a number.

So the first argument to the `itemRenderer` prop function, which is typically the item, needs to have an `id` property. I can see that information is missing from the [example at npm](https://www.npmjs.com/package/react-list-drag-and-drop). I'll try to add it as soon as I can.

**Comment by JulianG** [comment]
@liudonghua123 I have managed to reproduce your "Nothing was returned from render." issue when trying to drag an item which doesn't have an id property, so @eddiemarovich solution would probably work for you.

I have now added this to the documentation at v0.8.2, but I've also released v0.9.0 which throws an error when the `item` is not of type `object`, and when it doesn't contain an `id` property of type `number`.

![image](https://user-images.githubusercontent.com/237818/49968213-0e121280-ff1d-11e8-97b4-13c4d21b20dd.png)

I hope this helps, and again, apologies for having missed your initial message a month ago.

**Comment by eddiemarovich** [comment]
@JulianG As far as you know, would it be possible to have multiple `RLDD`s where you could drag and drop different items between them?

**Comment by liudonghua123** [comment]
@JulianG @eddiemarovich Thanks for you quickly replies, I will try and reply soon.

**Comment by liudonghua123** [comment]
Yes, I tried to add the `id` attribute to the data record then it worked. Thanks again. 😃

**Comment by JulianG** [comment]
> I recommend to add some tips about this, so anyone else could skip this problem.

@liudonghua123 I have added the requirements in the README.md which appears in github and npm as an HTML.

Also I've released a new version 0.9.0 which gives a more readable error when items don't contain an id property. 

Can you think of anything else we could do to help avoid this problem in the future?
